File: UI/financial_manager/src/app/models/token.ts
import { User } from './user';

export class Token {
  id?: number;
  refreshToken: string = '';
  isRevoked: boolean = false;
  expirationDate: Date;
  user?: User;

  constructor(refreshToken: string, isRevoked: boolean, expirationDate: Date) {
    this.refreshToken = refreshToken;
    this.isRevoked = isRevoked;
    this.expirationDate = expirationDate;
  }
}

File: financial-manager-api/Entities/Models/Token.cs
using System;
using System.Collections.Generic;
using financial_manager.Entities.Shared;

namespace financial_manager.Entities.Models;

public partial class Token : BaseModel
{
    public string RefreshToken { get; set; } = null!;

    public int UserId { get; set; }

    public DateTime ExpirationDate { get; set; }

    public bool IsRevoked { get; set; }

    public virtual User User { get; set; } = null!;
}
